vfs/dir.go

```diff
@@ -463,7 +493,10 @@ func (d *Dir) _readDir() error {
 		return err
 	}
 
+	go d.cacheCleanup()
```

Comment:
I'm not a big fan of running a go routine per directory read - these are going to mount up!

What you should probably do is use 

time.AfterFunc(d Duration, f func()) *Timer

And 
- keep the *Timer in the Dir structure
- kick the timer on every time the Dir is read with Timer.Reset()

Reply:
I like the `time.AfterFunc` method but want to verify my understanding once.

With the current implementation, wouldn't the goroutine only be created when the cache is actually stale? Otherwise, it returns from the `_read` function before it can create the goroutine.

Reply:
Done, thanks.